pkg/controller/utils/utils.go

```diff
@@ -19,26 +19,30 @@ const (
 	WaitTime  = 25
 )
 
-func MarshalIAMPolicy(role awsv1alpha1.AWSFederatedRole) (string, error) {
-	// The JSON tags as captials due to requirements for the policydoc
-	type awsStatement struct {
-		Effect    string                 `json:"Effect"`
-		Action    []string               `json:"Action"`
-		Resource  []string               `json:"Resource,omitempty"`
-		Principal *awsv1alpha1.Principal `json:"Principal,omitempty"`
-	}
+// The JSON tags as captials due to requirements for the policydoc
+type awsStatement struct {
+	Effect    string                       `json:"Effect"`
+	Action    []string                     `json:"Action"`
+	Resource  []string                     `json:"Resource,omitempty"`
+	Condition map[string]map[string]string `json:"Condition,omitempty"`
```

Comment:
Can I suggest that we implement a Condition type that will wrap this? This type is hard to parse visually.

It might be a good idea to just implement all of the possible elements of the JSON policy struct in AWS
https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements.html

Reply:
Hmm.  If we go that far I wonder if it would be worth it to just try to pull this entire package out and make it a standalone package, as we'd be just generalizing the entire IAM yaml -> json conversion.

Reply:
I'm :+1: to that. It doesn't look like AWS go sdk has any way to do this. But maybe better in a different PR. For this PR, I think at least making Condition its own type will help with readability

Reply:
I kinda hate the map syntax

Reply:
Yeah, that's fair.  I definitely wasn't a fan of trying to do the map of a map thing, but it was the quickest way to do it without having to declare the condition operator in the type.  I wonder if I can use the `json: -` thing to get that to work.  I'll test that out.